src/App.jsx

```diff
@@ -14,70 +17,65 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+export const App = () => {
+  const [selectedGood, setGood] = useState('Jam');
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  const isGoodSelected = good => selectedGood === good;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood
+          ? (
+            <>
+              {`${selectedGood} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => setGood('')}
+              />
+            </>
+          )
+          : (
+            'No goods selected'
+          )
+        }
+      </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              key={good}
+              data-cy="Good"
+              className={classNames({
+                'has-background-success-light': isGoodSelected(good),
+              })}
             >
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  data-cy={isGoodSelected(good)
+                    ? 'RemoveButton' : 'AddButton'}
+                  type="button"
+                  className={classNames('button', {
+                    'is-info': isGoodSelected(good),
+                  })}
+                  onClick={isGoodSelected(good)
+                    ? () => setGood('') : () => setGood(good)}
```

Comment:
```suggestion
                  onClick={isGoodSelected(good)
                    ? () => setGood('')
                    : () => setGood(good)}
```
Also, you can just path different values to the function conditionally instead of creating different functions.